WordPressComKit/SiteService.swift

```diff
@@ -1,44 +1,82 @@
 import Foundation
 import Alamofire
 
-public class SiteService {
-    public init() {}
-    
-    public func fetchSite(siteID: Int, completion: (Site?, NSError?) -> Void) {
-        Alamofire
-            .request(RequestRouter.Site(siteID: siteID))
-            .validate()
-            .responseJSON { response in
-                guard response.result.isSuccess else {
-                    completion(nil, response.result.error)
-                    return
-                }
-                
-                let json = response.result.value as? [String: AnyObject]
-                let site = self.mapJSONToSite(json!)
-                
-                completion(site, nil)
+extension Result {
+    func map<T>(transform: Value -> T) -> Result<T, Error> {
+        return flatMap({ .Success(transform($0)) })
+    }
+
+    func flatMap<T>(transform: Value -> Result<T, Error>) -> Result<T, Error> {
```

Comment:
Not sure I see the problem. Note that `T` might be different than `Value`. For a successful result, `flatMap` takes the value inside result and applies `transform` to it, which returns a new `Result` value.

If you mean't we're "creating" a new and different `.Failure` on error, I think that's both premature optimization and possibly not an issue. I believe enum is a compiler construct, but the runtime wouldn't do any extra allocations.